Add Host Pairs (#22038)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/security-hostpair.md

---
title: "hostPair resource type"

> description: "Represents a pair of parent and child hosts where the child-host was able to be reached via the parent-host"

Reviewer: ```suggestion
description: "Represents a pair of parent-child hosts in which the child host is reachable via the parent-host"
```

Author: Updated to reflect suggested change

> description: "Represents a pair of parent-child hosts in which the child host is reachable to be reached via the parent host"

Reviewer 2: Please clarify the description here; I'm not sure "reachable to be reached via" makes sense? Also make sure to add the missing period.

author: "jakedavies-microsoft"
ms.localizationpriority: medium
ms.prod: "security"
doc_type: resourcePageType
---

# hostPair resource type

Namespace: microsoft.graph.security

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

[!INCLUDE [threatintelligence-api-disclaimer](../../includes/threatintelligence-api-disclaimer.md)]

> Represents a pair of parent and child hosts where the child host was able to be reached via the parent host. For example, if contoso.com redirects to microsoft.com, then contoso.com would be the parent-host and microsoft.com the child-host.

Reviewer: ```suggestion
Represents a pair of parent-child hosts in which the child host is reachable via the parent host. For example, if contoso.com redirects to microsoft.com, then contoso.com would be the parent host and microsoft.com the child host.
```

Author: Updated to reflect suggested change

## Methods
|Method|Return type|Description|
|:---|:---|:---|
|[Get hostPair](../api/security-hostpair-get.md)|[microsoft.graph.security.hostPair](../resources/security-hostpair.md)|Read the properties and relationships of a **hostPair** object.|

> | [List childHostPairs for a host](../api/security-host-list-childhostpairs.md)|[microsoft.graph.security.hostPair](../resources/security-hostpair.md) collection|Get a list of **hostPair** objects associated with a host, where that host is the *parent* and has an outgoing pairing to a *child*.|

Reviewer: ```suggestion
|[List childHostPairs for a host](../api/security-host-list-childhostpairs.md)|[microsoft.graph.security.hostPair](../resources/security-hostpair.md) collection|Get a list of **hostPair** objects associated with a host, where that host is the *parent* and has an outgoing pairing to a *child*.|
```

|[List hostPairs for a host](../api/security-host-list-hostpairs.md)|[microsoft.graph.security.hostPair](../resources/security-hostpair.md) collection|Get a list of the [microsoft.graph.security.hostPair](../resources/security-hostpair.md) objects associated with a host, where the host is **either** the *parent* or *child*.|

> |[List parentHostPairs for a host](../api/security-host-list-hostpairs.md)|[microsoft.graph.security.hostPair](../resources/security-hostpair.md) collection|Get a list of the [microsoft.graph.security.hostPair](../resources/security-hostpair.md) objects associated with a host, where that host is the *child* and has an incoming pairing with a *parent*.|

Reviewer 3: For the Description fields - can you remove the full qualification, please? I have that wrong in the examples we're using

Author: Updated to reflect!

## Properties
|Property|Type|Description|
|:---|:---|:---|
|firstSeenDateTime|DateTimeOffset|The first date and time when Microsoft Defender Threat Intelligence observed the hostPair. The timestamp type represents date and time information using ISO 8601 format and is always in UTC. For example, midnight UTC on Jan 1, 2014, is 2014-01-01T00:00:00Z.|
|id|String|A system-generated ID for the hostPair.|
|lastSeenDateTime|DateTimeOffset|The last date and time when Microsoft Defender Threat Intelligence observed the hostPair. The timestamp type represents date and time information using ISO 8601 format and is always in UTC. For example, midnight UTC on Jan 1, 2014, is 2014-01-01T00:00:00Z.|

> |linkKind|String|The reason the two hosts were identified as hostPairs.|

Reviewer: ```suggestion
|linkKind|String|The reason the two hosts are identified as hostPairs.|
```
